test/object-store/sync/flx_sync.cpp

```diff
@@ -2864,17 +2903,19 @@ TEST_CASE("flx: data ingest", "[sync][flx][data ingest][baas]") {
             Object::create(c, realm, "Asymmetric", std::any(AnyDict{{"_id", foo_obj_id}, {"location", "foo"s}}));
             Object::create(c, realm, "Asymmetric", std::any(AnyDict{{"_id", bar_obj_id}, {"location", "bar"s}}));
             realm->commit_transaction();
-            wait_for_upload(*realm);
+
+            auto documents = get_documents(*realm->config().sync_config->user, "Asymmetric", 2);
```

Comment:
you should wait for upload before getting the documents. This works too because sync is not stopped, but you are querying the server unnecessarily (should be added in some other places below as well).

Reply:
Waiting for the upload to complete before polling doesn't really change how many times we poll. Upload completion happens about a quarter of the way through the time to when the documents are visible for asymmetric sync (and about 1% of the way for non-asymmetric).